Fix compile on CentOS7 gcc4.8.5

**Environment information**
* Operating System: CentOS7 x86_64
* Cutter version: master branch
* Obtained from:
  - [x] Built from source
  - [ ] Downloaded release from Cutter website or GitHub 
  - [ ] Distribution repository
* File format:

**Describe the bug**
compile error on CentOS7 with gcc4.8.5

<!-- A clear and concise description of what the bug is. -->

**To Reproduce**

Steps to reproduce the behavior:
1. follow the build instruction

**Expected behavior**
build without error

<!-- A clear and concise description of what you expected to happen. -->
https://stackoverflow.com/questions/30402943/has-cpp-attribute-not-a-function-like-macro
```
cutter/src/core/CutterCommon.h:65:56: error: missing binary operator before token "("
 #if defined(__has_cpp_attribute) && __has_cpp_attribute(deprecated)
                                                        ^
gmake[2]: *** [CMakeFiles/Cutter.dir/Cutter_autogen/mocs_compilation.cpp.o] Error 1
gmake[1]: *** [CMakeFiles/Cutter.dir/all] Error 2
gmake: *** [all] Error 2
```
this can be fixed as
```
#if defined(__has_cpp_attribute)
    #if __has_cpp_attribute(deprecated)
        #define DEPRECATED(msg) [[deprecated(msg)]]
    #endif
#else
#define CUTTER_DEPRECATED(msg)
#endif
```

**Screenshots**

<!-- If applicable, add screenshots to help explain your problem. -->


**Additional context**

<!-- Add any other context about the problem here. -->

--- a/src/core/CutterCommon.h
+++ b/src/core/CutterCommon.h
@@ -66,7 +66,8 @@ inline QString RHexString(RVA size)
     #if __has_cpp_attribute(deprecated)
         #define CUTTER_DEPRECATED(msg) [[deprecated(msg)]]
     #endif
-#else
+#endif
+#if !defined(CUTTER_DEPRECATED)
 #define CUTTER_DEPRECATED(msg)
 #endif
 

--- a/src/core/CutterCommon.h
+++ b/src/core/CutterCommon.h
@@ -62,8 +62,10 @@ inline QString RHexString(RVA size)
 #endif
 
 
-#if defined(__has_cpp_attribute) && __has_cpp_attribute(deprecated)
-#define CUTTER_DEPRECATED(msg) [[deprecated(msg)]]
+#if defined(__has_cpp_attribute)
+    #if __has_cpp_attribute(deprecated)
+        #define CUTTER_DEPRECATED(msg) [[deprecated(msg)]]
+    #endif
 #else
 #define CUTTER_DEPRECATED(msg)
 #endif

